python/cugraph/link_prediction/jaccard.py

```diff
@@ -108,13 +108,23 @@ def jaccard(input_graph, vertex_pair=None):
     if type(input_graph) is not Graph:
         raise Exception("input graph must be undirected")
 
-    # FIXME: Add support for multi-column vertices
     if type(vertex_pair) == cudf.DataFrame:
-        for col in vertex_pair.columns:
-            null_check(vertex_pair[col])
+        vertex_size = len(input_graph.renumber_map.implementation.col_names)
+        columns = vertex_pair.columns.to_list()
+        if vertex_size == 1:
+            for col in vertex_pair.columns:
+                null_check(vertex_pair[col])
+                if input_graph.renumbered:
+                    vertex_pair = input_graph.add_internal_vertex_id(
+                        vertex_pair, col, col
+                    )
+        else:
             if input_graph.renumbered:
                 vertex_pair = input_graph.add_internal_vertex_id(
-                    vertex_pair, col, col
+                    vertex_pair, "src", columns[:vertex_size]
+                )
+                vertex_pair = input_graph.add_internal_vertex_id(
+                    vertex_pair, "dst", columns[vertex_size:]
```

Comment:
This pattern is repeated in several algorithms - can the entire if-elif-else block be replaced by a utility that can be reused everywhere? This would remove a lot of code and make what's happening more self-documenting IMO, and could also minimize the amount of places that would need to be updated when implementation changes.  Maybe something that can be used like:
```
if type(input_graph) is not Graph:
    raise Exception("input graph must be undirected")

vertex_pairs = prepare_vertex_pairs(input_graph, vertex_pairs)

df = jaccard_wrapper.jaccard(input_graph, None, vertex_pairs)
```
side note: it seems like an algo call shouldn't modify the graph like this (eg. adding internal vertex IDs to the graph), but that's probably a bigger problem for a different PR.